src/mds/Server.cc

```diff
@@ -2318,6 +2332,13 @@ void Server::handle_client_request(const cref_t<MClientRequest> &req)
 	mds->queue_one_replay();
       return;
     }
+
+    std::string_view fs_name = mds->get_fs_name();
+    if (!fs_name.empty() && !session->fs_name_capable(fs_name, MAY_READ)) {
+      dout(0) << "unauthorized request from " << req->get_source()
+	      << ", dropping" << dendl;
+      return;
+    }
```

Comment:
There's no need for this check. We will do the check when the session is opened.

Reply:
Ok. Getting rid of this...